experimental/Schemes/src/BlowupMorphism.jl

```diff
@@ -644,13 +644,17 @@ function produce_object_on_affine_chart(I::StrictTransformIdealSheaf, U::AbsAffi
   f_loc = covering_morphism(f)[U]
   V = codomain(f_loc)
   IE_loc = IE(U)
+  @assert isone(ngens(IE_loc)) "ideal sheaf of exceptional locus is not principal"
   tot = pullback(f_loc)(J(V))
   #return saturation_with_index(tot, IE_loc)
   # It is usually better to pass to the simplified covering to do the computations
   simp_cov = simplified_covering(X)
   U_simp = first([V for V in patches(simp_cov) if original(V) === U])
   a, b = identification_maps(U_simp)
-  result, _ = saturation_with_index(pullback(a)(tot), pullback(a)(IE_loc))
+  # This used to be the following line. But we don't use the index, so we 
+  # switch to the more performant version
+  # result, _ = saturation_with_index(pullback(a)(tot), pullback(a)(IE_loc))
+  result = _iterative_saturation(pullback(a)(tot), elem_type(OO(U_simp))[pullback(a)(u) for (u, _) in factor(lifted_numerator(first(gens(IE_loc))))])
```

Comment:
@afkafkafk13 : The use of  `saturation_with_index` here goes back to your original code for computing transforms. Do you see a reason why we should not switch to `_iterative_saturation` here? It is probably more performant in many cases.

Reply:
You are perfectly right here. In the very first version of the code, I assumed a single blow up at an irreducible component of the center -- for which the saturation_with_index was fine (and allowed extra debugging information). In sequences of blow ups this should indeed be done iteratively. The overall structure has evolved very far over time.

Reply:
Just one more thought: Is the exceptional divisor reliably stored as principal on the given chart? We do not have full control on this.
I would not put the condition in an `assert` up there, but rather use the old variant as a fallback for exceptional divisors which happen to be stored non-principal.

Reply:
Ok, sounds reasonable. Thx.